### PR TITLE
refactor: erroring on raw response (deployer)

### PR DESCRIPTION
**Pull request by chesedo**
Fixes: ENG-57

### CODE REVIEW

File: cargo-shuttle/src/client.rs [hunk 1]
```diff
@@ -33,11 +34,12 @@ pub(crate) async fn auth(mut api_url: ApiUrl, username: String) -> Result<ApiKey
...
 
-    Err(anyhow!(
-        "status: {}, body: {}",
-        response_status,
-        response_text
-    ))
+    error!(
+        text = response_text,
+        status = %response_status,
+        "failed to authenicate with server"
```
Comment (brokad) [inline cargo-shuttle/src/client.rs:40]:
```suggestion
        "failed to authenticate with server"
```

File: cargo-shuttle/src/client.rs [hunk 2]
```diff
@@ -220,6 +223,12 @@ async fn to_api_result(res: Response) -> Result<DeploymentMeta> {
     let text = res.text().await?;
     match serde_json::from_str::<DeploymentMeta>(&text) {
         Ok(meta) => Ok(meta),
-        Err(_) => Err(anyhow!("{}", text)),
+        Err(error) => {
+            error!(
+                error = &error as &dyn std::error::Error,
+                text, "failed to parse deployment meta"
+            );
+            Err(anyhow!("could not parse server response"))
```
Comment (brokad) [inline cargo-shuttle/src/client.rs:231]: Does using `error.context(anyhow!(...))` provide a nicer output?

File: cargo-shuttle/Cargo.toml [hunk 3]
```diff
@@ -31,6 +30,8 @@ structopt = "0.3.26"
 tokio = "1.19.2"
 toml = "0.5.9"
 toml_edit = "0.14.4"
+tracing = "0.1.35"
```
Comment (brokad) [inline cargo-shuttle/Cargo.toml:33]: `tracing` is taking over 😉

File: cargo-shuttle/src/client.rs [hunk 4]
```diff
@@ -87,11 +89,12 @@ pub(crate) async fn shuttle_version(mut api_url: ApiUrl) -> Result<String> {
...
-            "status: {}, body: {}",
-            response_status,
-            res.text().await?
-        ))
+        error!(
+            text = res.text().await?,
+            status = %response_status,
+            "failed to get shuttle version from server"
+        );
+        Err(anyhow!("failed to get shuttle version from server",))
```
Comment (brokad) [inline cargo-shuttle/src/client.rs:97]: Same question as a comment below?

**Review by brokad — APPROVED** [review]

File: cargo-shuttle/src/client.rs (hunk 2 above)
Comment (chesedo) [reply cargo-shuttle/src/client.rs:231]: Yip, it gives a bit more of a "backtrace". Thanks!

File: cargo-shuttle/src/client.rs (hunk 4 above)
Comment (chesedo) [reply cargo-shuttle/src/client.rs:97]: Not quite, there is no `Result` here to call `.context()` on - ie nothing to "backtracify" so nothing to make nicer

**Pull request merged**